app/src/ui/dialog/dialog.tsx

```diff
@@ -106,6 +107,13 @@ interface IDialogState {
    * grace period or not.
    */
   readonly isAppearing: boolean
+
+  /**
+   * An optional id for the h1 element that contains the title of this
+   * dialog. Used to aid in accessibility. Undefined if the dialog does
+   * not have a title or the component has not yet been mounted.
+   */
+  readonly titleId?: string
```

Comment:
Addressed this in the documentation but the aria system only supports referring to other elements by id.